src/server/rdb_save.cc

```diff
@@ -1197,15 +1197,22 @@ error_code RdbSaver::SaveAux(const StringVec& lua_scripts) {
   RETURN_ON_ERR(SaveAuxFieldStrInt("redis-bits", 64));
 
   RETURN_ON_ERR(SaveAuxFieldStrInt("ctime", time(NULL)));
-
   RETURN_ON_ERR(SaveAuxFieldStrInt("used-mem", used_mem_current.load(memory_order_relaxed)));
-
   RETURN_ON_ERR(SaveAuxFieldStrInt("aof-preamble", aof_preamble));
 
   // Save lua scripts only in rdb or summary file
-  DCHECK(save_mode_ != SaveMode::SINGLE_SHARD || lua_scripts.empty());
-  for (const string& s : lua_scripts) {
+  DCHECK(save_mode_ != SaveMode::SINGLE_SHARD || glob_state.lua_scripts.empty());
+  for (const string& s : glob_state.lua_scripts)
     RETURN_ON_ERR(impl_->SaveAuxFieldStrStr("lua", s));
+
+  if (save_mode_ == SaveMode::RDB) {
```

Comment:
Nice! So that was a bug!

Reply:
No, I just print a warnings now 😄 Unknown aux fields are ignored either way